packages/react-components/src/components/DomainPermissions.tsx

```diff
@@ -84,14 +88,18 @@ const DomainPermissionsContainer = (props: DomainPermissionsProps): JSX.Element
         </Stack>
       </Stack>
       <Stack styles={textContainerStyles}>
-        <Text styles={primaryTextStyles}>{_formatString(strings.primaryText, { appName: appName })}</Text>
-        <Text styles={secondaryTextStyles}>{strings.secondaryText}</Text>
+        {strings && isValidString(strings?.primaryText) && (
+          <Text styles={primaryTextStyles}>{_formatString(strings.primaryText, { appName: appName })}</Text>
+        )}
+        <Text styles={secondaryTextStyles}>{strings?.secondaryText}</Text>
         {onAllowAccessClick && (
-          <PrimaryButton styles={primaryButtonStyles} text={strings.primaryButtonText} onClick={onAllowAccessClick} />
+          <PrimaryButton styles={primaryButtonStyles} text={strings?.primaryButtonText} onClick={onAllowAccessClick} />
```

Comment:
You will need the same `strings && isValidString` for each of these, otherwise there will be an empty button with no text here (and apply the same to the secondary text element above)